### PR TITLE
Add access for Dagster to xPro database

### PR DESCRIPTION
**Pull request by blarghmatey**
The switch to the new database instance for xPro changed the access list, preventing Dagster from connecting. This PR updates Dagster with a dedicated security group and grants access for it to the xPro RDS instance.

### CODE REVIEW

File: src/ol_infrastructure/applications/dagster/__main__.py [hunk 1]
```diff
@@ -48,15 +49,30 @@
     "Statement": [
         {
             "Effect": "Allow",
-            "Action": ["s3:List*", "s3:Get*", "s3:Put*", "s3:Delete*"],
+            "Action": "s3:ListAllMyBuckets",
+            "Resource": "*",
```
Comment (shaidar) [inline src/ol_infrastructure/applications/dagster/__main__.py:53]: Why does it need list access to all buckets?

File: src/ol_infrastructure/infrastructure/operations/consul.py [hunk 2]
```diff
@@ -52,7 +51,6 @@
 consul_instance_profile = iam.InstanceProfile(
     f"consul-instance-profile-{environment_name}",
     role=consul_instance_role.name,
-    name=f"consul-instance-profile-{environment_name}",
     path="/ol-operations/consul/profile/",
```
Comment (shaidar) [inline src/ol_infrastructure/infrastructure/operations/consul.py:54]: Why the changes here?

Comment (blarghmatey) [reply src/ol_infrastructure/infrastructure/operations/consul.py:54]: Because it was giving me an error when trying to replace the resource due to name conflicts.
